packages/client/hmi-client/src/composables/useCharts.ts

```diff
@@ -468,7 +468,20 @@ export function useCharts(
 					statLayerVariables.push(`${chartData.value?.pyciemssMap[v]}_mean:pre`);
 				}
 			});
-			colorScheme.push(...CATEGORICAL_SCHEME);
+			if (!setting.variableColors || setting.variableColors === undefined) {
+				setting.variableColors = {};
+			}
+			const variableColors = {};
+			variables.forEach((variable, index) => {
+				if (setting.variableColors?.[variable] && !isEmpty(setting.variableColors[variable])) {
+					colorScheme.push(setting.variableColors[variable]);
+				} else {
+					variableColors[variable] = CATEGORICAL_SCHEME[index % CATEGORICAL_SCHEME.length];
+					colorScheme.push(variableColors[variable]);
+				}
+			});
+			Object.assign(setting.variableColors, variableColors);
+			setting.colorScheme = [...colorScheme];
```

Comment:
Yeah I would like to avoid modifying setting object directly here, is there other work around?